src/features/l2ps-messaging/L2PSMessagingServer.ts

```
@@ -161,31 +169,31 @@ export class L2PSMessagingServer {
             return
         }
 
-        // Remove old connection if re-registering
-        const existing = this.peers.get(publicKey)
+        // Remove old connection if re-registering (canonical identity)
+        const existing = this.peers.get(canonicalKey)
         if (existing) {
             try { (existing.ws as ServerWebSocket<WSData>).close() } catch {}
```

ESLint

[error] Empty block statement.

(no-empty)